dotnet/targets/dotnet-xcsync.targets

```diff
@@ -0,0 +1,72 @@
+			<XcsyncCommonArguments Condition="'$(XcsyncVerbosity)' != ''">$(XcsyncCommonArguments) -v $(XcsyncVerbosity)</XcsyncCommonArguments>
+			<XcsyncCommonArguments>$(XcsyncCommonArguments) --dotnet-path "$([System.IO.Path]::GetDirectoryName($(DOTNET_HOST_PATH)))"</XcsyncCommonArguments>
+
+			<XcsyncGenerateArguments>$(XcsyncCommonArguments) --force</XcsyncGenerateArguments>
+			<XcsyncGenerateArguments Condition="'$(XcsyncOpenXcode)' != 'false'">$(XcsyncGenerateArguments) --open</XcsyncGenerateArguments>
+
+			<XcsyncSyncArguments>$(XcsyncCommonArguments)</XcsyncSyncArguments>
+		</PropertyGroup>
+	</Target>
+
+	<Target Name="CheckXcsyncInstalled">		
+		<PropertyGroup>
+			<LocalErrorCode>0</LocalErrorCode>
+			<GlobalErrorCode>0</GlobalErrorCode>
+			<XcsyncInstalled>False</XcsyncInstalled>	
+		</PropertyGroup>
+	    <Exec ContinueOnError="True" IgnoreStandardErrorWarningFormat="true" Command="$(XcsyncExecute)" StandardErrorImportance="Low" StandardOutputImportance="Low">
+       		<Output TaskParameter="ExitCode" PropertyName="GlobalErrorCode"/>
+    	</Exec>
+	    <Exec ContinueOnError="True" IgnoreStandardErrorWarningFormat="true" Command="$(DOTNET_HOST_PATH) tool run xcsync" Condition="'$(GlobalErrorCode)' != '0'" StandardErrorImportance="Low" StandardOutputImportance="Low">
+       		<Output TaskParameter="ExitCode" PropertyName="LocalErrorCode"/>
+    	</Exec>
+		<PropertyGroup>
+			<XcsyncInstalled Condition="'$(GlobalErrorCode)' == '0' OR '$(LocalErrorCode)' == '0'">True</XcsyncInstalled>
+			<XcsyncInstalled Condition="'$(GlobalErrorCode)' != '0' AND '$(LocalErrorCode)' != '0'">False</XcsyncInstalled>		
+			<XcsyncExecute Condition="'$(GlobalErrorCode)' != '0' AND '$(LocalErrorCode)' == '0'">$(DOTNET_HOST_PATH) tool run xcsync</XcsyncExecute>
+		</PropertyGroup>
+		<Message Importance="High" Text="The xcsync tool is not installed." Condition="'$(XcsyncInstalled)' != 'False'" />
+	</Target>
+
+	<Target Name="InstallXcsync" DependsOnTargets="CheckXcsyncInstalled">
+		<PropertyGroup>
+			<InstallErrorCode>0</InstallErrorCode>
+		</PropertyGroup>
+		<Message Importance="High" Text="Installing xcsync tool..." Condition="'$(XcsyncInstalled)' == 'False'"/>
+	    <Exec ContinueOnError="True" IgnoreStandardErrorWarningFormat="true" Command="$(DOTNET_HOST_PATH) tool install -g dotnet-xcsync --prerelease" Condition="'$(XcsyncInstalled)' == 'False'" StandardErrorImportance="Low" StandardOutputImportance="Low">
+       		<Output TaskParameter="ExitCode" PropertyName="InstallErrorCode"/>
+    	</Exec>
+		<Error Text="The xcsync tool could not be installed. Please install xcsync by running 'dotnet tool install -g dotnet-xcsync --prerelease'." Condition="'$(InstallErrorCode)' != '0'" />
+		<Error Text="The xcsync tool was installed globally. Remove it using the following command: 'dotnet tool install -g dotnet-xcsync --prerelease'." Condition="'$(InstallErrorCode)' == '0'" />
```

Comment:
Won't this always fail? Either one or the other error will always be shown.

Reply:
Somehow missed some changes in the last commit, fixed.